ArduCopter/Copter.cpp

```diff
@@ -772,13 +772,15 @@ void Copter::update_simple_mode(void)
 {
     float rollx, pitchx;
 
-    // exit immediately if no new radio frame or not in simple mode
-    if (simple_mode == SimpleMode::NONE || !ap.new_radio_frame) {
+    // exit immediately if not in simple mode:
+    if (simple_mode == SimpleMode::NONE) {
         return;
     }
 
-    // mark radio frame as consumed
-    ap.new_radio_frame = false;
+    // exit immediately if RC input is invalid:
+    if (!rc().has_valid_input()) {
```

Comment:
This is not the same; RC input would come in at maybe 50Hz, this method is called at the main loop rate, so we'd start to do these transforms much faster than required in the case someone's using simple mode.  The method is also not idempotent, so pretty much completely broken.

We probably want to unconditionally call this method when a new radio frame is received instead.